specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,422 @@
+Programmatic Save As API
+===
+
+# Background
+
+Chromium browser's context menus have a "Save as" menu item to save the top level 
+document (html page, image, pdf, or other content) through a save as dialog. We 
+provide more flexible ways to programmatically perform the Save As operation in 
+WebView2. 
+
+With the new API you will be able to:
+- Launch the default save as dialog
+- Block the default save as dialog
+- Request save as silently by providing the path and save as kind
+- Build your own save as UI
+
+The chromium browser's Save As operation consists of showing the Save As dialog 
+and then starting a download of the top level document. The Save As method and 
+event described in this document relate to the Save As dialog and not the download, 
+which will go through the existing WebView2 download APIs. 
+
+We'd appreciate your feedback.
+
+# Description
+
+We propose the `RequestSaveAs` method WebView2, which allows you to trigger 
+the Save As UX programmatically. By using this method, the system default dialog, 
+or your own UI will show and start the Save As operation.
+
+We also propose the `CoreWebView2.SaveAsRequested` event. You can register this event to block 
+the default dialog and instead create your own Save As UI using the `SaveAsRequestedEventArgs`, 
+to set your preferred save as path, save as kind, and duplicate file replacement rule. 
+In your client app, you can design your own UI to input these parameters. 
+For HTML documents, we support 3 save as kinds: HTML_ONLY, SINGLE_FILE and 
+COMPLETE. Non-HTML documents, you must use DEFAULT, which will save the content as 
+it is. This API has default values for all parameters, to perform the common 
```

Comment:
What does DEFAULT produce for HTML documents?

Reply:
Correct. Ensure this is covered at least in the ref docs.